packages/grafana-ui/src/components/Forms/Form.story.tsx

```diff
@@ -74,34 +73,38 @@ const renderForm = (defaultValues?: Partial<FormDTO>) => (
           <Legend>Edit user</Legend>
 
           <Field label="Name" invalid={!!errors.name} error="Name is required">
-            <Input name="name" placeholder="Roger Waters" ref={register({ required: true })} />
+            <Input {...register('name', { required: true })} placeholder="Roger Waters" />
           </Field>
 
           <Field label="Email" invalid={!!errors.email} error="E-mail is required">
-            <Input id="email" name="email" placeholder="roger.waters@grafana.com" ref={register({ required: true })} />
+            <Input {...register('email', { required: true })} id="email" placeholder="roger.waters@grafana.com" />
           </Field>
 
           <Field label="Username">
-            <Input name="username" placeholder="mr.waters" ref={register} />
+            <Input {...register('username')} placeholder="mr.waters" />
           </Field>
           <Field label="Nested object">
-            <Input name="nested.path" placeholder="Nested path" ref={register} />
+            <Input {...register('nested.path')} placeholder="Nested path" />
```

Comment:
out of curiosity, is the register path argument typed?

Reply:
Afaik the types come from the default values passed to the form or fields.